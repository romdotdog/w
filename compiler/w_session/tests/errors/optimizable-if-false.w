fn main(a: u32, b: u32): u32 {
    if 0 {
        a
    } else {
        b
    }
}

// more complex case
fn bar(a: u32, b: u32): u32 {
    let c = 1;
    if c ^ c {
        a
    } else {
        b
    }
}