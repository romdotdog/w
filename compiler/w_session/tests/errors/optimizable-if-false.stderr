tests/errors/optimizable-if-false.w:2:8: error: condition evaluates to false
    if 0 {
       ^
tests/errors/optimizable-if-false.w:12:8: error: condition evaluates to false
    if c ^ c {
       ^^^^^
